### ISSUE TITLE
min() with overloads giving weird errors

### ISSUE THREAD

**Issue by BenjyRead** [microsoft/pyright#9306]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**

How to reproduce:
It *should* be just install pyright in neovim (I used mason) and type in the following code in a empty python file:
```python
d = {'a': 1, 'b': 2}
print(min(d, key=d.get))
``` 

python version is 3.12.7 in the screenshot

**Code or Screenshots**
![Image](https://github.com/user-attachments/assets/aaadbea5-9d92-4f45-8f0c-0bb980c70d78)

```python
d = {'a': 1, 'b': 2}
print(min(d, key=d.get))
```

**VS Code extension or command-line**
I am running pyright as an LSP in neovim, installed using mason (on lazy.nvim)

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

The problem is that `dict.get` is defined to return a value of `int | None`, and the `min` function doesn't accept such a callback because it can't handle a `None` response.
